src/Tizen.NUI/src/public/Utility/DpTypeConverter.cs

```diff
@@ -0,0 +1,104 @@
+{
+    /// <summary>
+    /// Default DpTypeConverter class to convert dp types.
+    /// </summary>
+    /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public sealed class DpTypeConverter : GraphicsTypeConverter
+    {
+        private volatile static DpTypeConverter dpTypeConverter;
+
+        /// <summary>
+        /// An unique Singleton Instance of DpTypeConverter
+        /// </summary>
+        /// <value>Singleton instance of DpTypeConverter</value>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static DpTypeConverter Instance
+        {
+            get
+            {
+                if (dpTypeConverter == null)
+                {
+                    dpTypeConverter = new DpTypeConverter();
+                }
+
+                return dpTypeConverter;
+            }
+        }
+
+        /// <summary>
+        /// Converts script to px
+        /// </summary>
+        /// <returns>Pixel value that is converted from input string</returns>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override float ConvertScriptToPixel(string scriptValue)
+        {
+            float convertedValue = 0;
+            if (scriptValue != null)
+            {
+                if (scriptValue.EndsWith("dp"))
+                {
+                    convertedValue = ConvertToPixel(float.Parse(scriptValue.Substring(0, scriptValue.LastIndexOf("dp")), CultureInfo.InvariantCulture));
+                }
+                else if (scriptValue.EndsWith("px"))
+                {
+                    convertedValue = float.Parse(scriptValue.Substring(0, scriptValue.LastIndexOf("px")), CultureInfo.InvariantCulture);
+                }
+                else
+                {
+                    if (!float.TryParse(scriptValue, NumberStyles.Any, CultureInfo.InvariantCulture, out convertedValue))
+                    {
+                        NUILog.Error("Cannot convert the script {scriptValue}\n");
+                        convertedValue = 0;
+                    }
+                }
+            }
+            return convertedValue;
+        }
+
+        /// <summary>
+        /// Converts dp type to px
+        /// </summary>
+        /// <returns>Pixel value that is converted by the the display matric</returns>
+        /// This will be public opened in tizen_next after ACR done. Before ACR, need to be hidden as inhouse API.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override float ConvertToPixel(float value)
+        {
+            return value * (GraphicsTypeManager.Instance.Dpi / (float)GraphicsTypeManager.Instance.DefaultDpi);
```

Comment:
제 생각으로는 ScaledDpi를 만든 이유가 DPI기준으로 고정된 변환을 하려고 하는것이 아니라 ScaleFactor에 의해서 변환된 기준으로 변환을 하기 위해서 만들어놓은것 같은데요, 그러면 그 변환도 그 기준에 맞춰서 해야지 의미가 있을듯합니다.

즉 ScaledDpi와 Density가 의미하는 바와 다른 결과로 계산을 하면 안될듯합니다.

```suggestion
            return value * (GraphicsTypeManager.Instance.ScaledDpi / (float)GraphicsTypeManager.Instance.DefaultDpi);
```